internal/error_test.go

```diff
@@ -812,3 +832,35 @@ func Test_convertFailureToError_ServerFailure(t *testing.T) {
 	require.Equal("message", serverErr.Error())
 	require.Equal(true, serverErr.nonRetryable)
 }
+
```

Comment:
Add tests for the chained errors and failures.

Reply:
There is `Test_convertErrorToFailure_ActivityTaskError` and `Test_convertErrorToFailure_ChildWorkflowExecutionError`. I will add chains to new tests also.